Common/Extensions.cs

```diff
@@ -1050,17 +1050,31 @@ public static bool TryConvert<T>(this PyObject pyObject, out T result)
             {
                 try
                 {
+                    if (typeof(T) == typeof(Type))
+                    {
+                        result = pyObject.As<Type>() as T;
+                        return true;
+                    }
+
+                    var pythonType = pyObject.GetPythonType();
```

Comment:
This call constitutes a round-trip into Python to get the type as a PyObject which we don't even use in the following two tests.

I still maintain that the code makes no sense, but at least it could be a little faster at calculating the wrong answer.